src/App.tsx

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+
+import { getTodos } from './api';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [status, setStatus] = useState('');
+
+  const getTodosFromServer = useCallback(async () => {
+    const todosFromServer = await getTodos();
+
+    setTodos(todosFromServer);
+    setIsTodosLoaded(true);
+  }, []);
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const handleTodoSelection = useCallback((todo: Todo) => {
+    setSelectedTodo(todo);
+  }, []);
+
+  const handleRemoveTodoSelection = useCallback(() => {
+    setSelectedTodo(null);
+  }, []);
+
+  const handleSetQuery = useCallback((value: string) => {
+    setQuery(value);
+  }, []);
+
+  const handleSetStatus = useCallback((value: string) => {
+    setStatus(value);
+  }, []);
+
+  const filterTodos = () => {
+    let filteredTodos;
+
+    switch (status) {
+      case Status.Active:
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+      case Status.Completed:
+        filteredTodos = todos.filter(todo => todo.completed);
```

Comment:
You could just set in cases a callback by witch you will filter to not write filtering everytime but thats ok